### PR TITLE
Gcp gke docs

### PR DESCRIPTION
**Pull request by mastoj**
Adds documentation for: https://github.com/SAFE-Stack/SAFE-template/pull/209

Depends on: https://github.com/SAFE-Stack/docs/pull/90

The dependency is there to re-use the general parts on google cloud.

### CODE REVIEW

**Comment by theimowski** [comment]
This one is definitely more complex than app engine one.
I got following error when trying to add a cluster:

```
gcloud container clusters create safe-cluster --num-nodes=3 -m f1-micro --region europe-west3
WARNING: Starting in 1.12, new clusters will have basic authentication disabled by default. Basic authentication can be enabled (or disabled) manually using the `--[no-]enable-basic-auth` flag.
WARNING: Starting in 1.12, new clusters will not have a client certificate issued. You can manually enable (or disable) the issuance of the client certificate using the `--[no-]issue-client-certificate` flag.
WARNING: Currently VPC-native is not the default mode during cluster creation. In the future, this will become the default mode and can be disabled using `--no-enable-ip-alias` flag. Use `--[no-]enable-ip-alias` flag to suppress this warning.
WARNING: Starting in 1.12, default node pools in new clusters will have their legacy Compute Engine instance metadata endpoints disabled by default. To create a cluster with legacy instance metadata endpoints disabled in the default node pool, run `clusters create` with the flag `--metadata disable-legacy-endpoints=true`.
This will enable the autorepair feature for nodes. Please see https://cloud.google.com/kubernetes-engine/docs/node-auto-repair for more information on node autorepairs.
WARNING: Starting in Kubernetes v1.10, new clusters will no longer get compute-rw and storage-ro scopes added to what is specified in --scopes (though the latter will remain included in the default --scopes). To use these scopes, add them explicitly to --scopes. To use the new behavior, set container/new_scopes_behavior property (gcloud config set container/new_scopes_behavior true).
ERROR: (gcloud.container.clusters.create) ResponseError: code=403, message=
	(1) insufficient regional quota to satisfy request: resource "CPUS": request requires '9.0' and is short '1.0'. project has a quota of '8.0' with '8.0' available. View and manage quotas at https://console.cloud.google.com/iam-admin/quotas?usage=USED&project=safe-template-test
	(2) insufficient regional quota to satisfy request: resource "IN_USE_ADDRESSES": request requires '9.0' and is short '1.0'. project has a quota of '8.0' with '8.0' available. View and manage quotas at https://console.cloud.google.com/iam-admin/quotas?usage=USED&project=safe-template-test.
```

Maybe you can't create a cluster with a free account?

**Comment by mastoj** [comment]
I think you should be able to create it if you have some of the free $300
credits still left. If you don't have that it might not be possible. Hard
for me to test, I can check what I can do.

**Comment by mastoj** [comment]
I did some quick experimation @theimowski, and here is what I concluded on. The error you see is probably because you are using the resources that are limited to your account, I guess there are some default limits so you don't waste all your money. This can be connected to your previous work with the appengine stuff, if you tried multiple deploys since I think new deploys still keep the old deploy around if not explicitly told to delete.

I'm quite sure it is not related to billing/free account, since there is no such thing as a free account. There are some free spending, but not free account. I creataed a new project where I disconnected the billing completely, and that gave some other errors than what you saw. After enabling billing again I could create the cluster with the command that you see. I think that cluster should create a really cheap cluster since the compute resources for those node types should have some free hours connected to them.

**Comment by theimowski** [comment]
I was able to create the cluster with 2 nodes - is it enough or does one need 3 to start with?

File: docs/template-gcp-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+[Kubernetes Engine](https://cloud.google.com/kubernetes-engine/) is a managed, production-ready environment for deploying containerized applications on Google Cloud Platform.
+
+Compared to AppEngine you will get faster deploy, but it also comes at a cost of higher operational cost. Total cost might be lower though depending on how many apps you run and how large your cluster is. One cluster can often host multiple apps, while each AppEngine app requires a single machine so with many apps AppEngine might be more expensive.
+
+Covering Kubernetes and Kubernetes Engine in greater detail is too large of a topic in this context, and there are a lot of tutorials on those topics available.
+
+## Deployment steps
+
+Before you can deploy your AppEngine app you need to create a [Google Cloud Platform account](template-google-cloud.md#creating-an-account), [set up the `gcloud` CLI tool]((template-google-cloud.md#setup-gcloud-cli-tool)) and [enable billing](template-google-cloud.md#enable-billing).
```
[the remark this answers is not in the document]
Comment (mastoj) [reply docs/template-gcp-kubernetes.md]: Fixed

File: docs/template-gcp-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+The SAFE template has the ability to deploy to [Kubernetes Engine](https://cloud.google.com/kubernetes-engine/) on Google Cloud Platform, GCP, with minimum effort. The focus of the template is to set up the basic building blocks like creating the container, publishing the container and also create a simple deployment on kubernetes that exposes a kubernetes service. You probably want to, and should, add things like a proper hostname and https support. That can be done in multiple ways and is more of a kubernetes specific task than a SAFE task.
+
+## Quickstart
```
[the remark this answers is not in the document]
Comment (mastoj) [reply docs/template-gcp-kubernetes.md]: Added commands to create a git repo with one commit.

File: docs/template-gcp-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+## Deployment steps
+
+Before you can deploy your AppEngine app you need to create a [Google Cloud Platform account](template-google-cloud.md#creating-an-account), [set up the `gcloud` CLI tool]((template-google-cloud.md#setup-gcloud-cli-tool)) and [enable billing](template-google-cloud.md#enable-billing).
+
+### Creating your cluster and installing kubectl
+
+Next up you need to create your kubernetes cluster. To create a cluster where the nodes are using of the free tier computing resources you can use this command:
+
+```
+gcloud container clusters create safe-cluster --num-nodes=3 -m f1-micro
```
[the remark this answers is not in the document]
Comment (mastoj) [reply docs/template-gcp-kubernetes.md]: The easiest way to deal with this is to have a default environment configured, so I added some text on how to set that up.

File: docs/template-gcp-kubernetes.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mastoj) [reply docs/template-gcp-kubernetes.md]: Removed extra ( ).

**Comment by mastoj** [comment]
@theimowski I really don't understand how you manage to create with only two nodes. What do you get if you run `kubectl get nodes`? When I create a cluster of `f1-micro` instances I get this message: 

> ERROR: (gcloud.container.clusters.create) ResponseError: code=400, message=Clusters of f1-micro instances must contain at least 3 nodes. Please make the cluster larger or use a different machine type.

So I think we should keep 3.

**Comment by theimowski** [comment]
```
$ kubectl get nodes
NAME                                          STATUS    ROLES     AGE       VERSION
gke-safe-cluster-default-pool-07431625-758n   Ready     <none>    1d        v1.10.9-gke.5
gke-safe-cluster-default-pool-07431625-bt3k   Ready     <none>    1d        v1.10.9-gke.5
gke-safe-cluster-default-pool-313aac99-9lft   Ready     <none>    1d        v1.10.9-gke.5
gke-safe-cluster-default-pool-313aac99-q5x3   Ready     <none>    1d        v1.10.9-gke.5
gke-safe-cluster-default-pool-f84888e0-49c9   Ready     <none>    1d        v1.10.9-gke.5
gke-safe-cluster-default-pool-f84888e0-vwcr   Ready     <none>    1d        v1.10.9-gke.5
```

So it seems it multiplies times 3 in my case for some reason 🤷‍♂️  When I did with `--num-nodes=3` I got the quota error that I only have 8 CPUs to use while the request needs **9**

Anyway, I finally managed to get it working after those couple amends, and successfully deployed SAFE template to kubernetes engine, which is pretty neat.

One note, during build script I got:
```
.> "kubectl" get deploy safe-template (In: false, Out: false, Err: false)
Error from server (NotFound): deployments.extensions "safe-template" not found
```
The script finished OK, but we should address this as well

**Comment by mastoj** [comment]
@theimowski that's really odd behavior there, I'm not that seasoned in kubernetes to say what's going on. `--num-nodes` should of course give you the number of nodes you need, and you need at least 3.

Ah ok, you get the first time when the script tries to figure out if your deployment exists or not. I can probably solve that differently by doing a `kubectl get deploy` (without the app name) and then check the result if it contains the app.

**Comment by mastoj** [comment]
Everything is fixed in the last commit to the template.

**Comment by theimowski** [comment]
Can you rebase on master?

**Comment by mastoj** [comment]
Rebased this yesterday.

File: docs/template-gcp-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+## Quickstart
+
+If you are familiar with GCP and already have `gcloud` and `kubectl` configured and an existing kubernetes cluster all you need to do to get started is running these commands:
+
+```
+git init
+git add .
+git commit -m "Initial commit"
+dotnet new SAFE --deploy gcp-kubernetes
+fake build -t Deploy -e SAFE_CLUSTER=<kubernetes cluster>
```
Comment (theimowski) [inline docs/template-gcp-kubernetes.md]: Let's remove the additional SAFE_CLUSTER variable from this invocation, and let it default to `safe-cluster`. We can add a paragraph on SAFE_CLUSTER optional variable below

**Comment by mastoj** [comment]
@theimowski I think all the comments have been addressed now, both in the template and in the docs.

**Comment by theimowski** [comment]
Thanks again @mastoj - this is really awesome 👍  Will be released in v0.45

**Comment by mastoj** [comment]
No problem @theimowski :). With this in step it should be super simple to add kubernetes support for any provider, what most likely will differ is where you push the image basically.

**Comment by mastoj** [comment]
I noticed that it was released, but the published docs doesn't reflect the. Isn't the docs automatically published when merged to master?

**Comment by theimowski** [comment]
nope they need to be manually pushed to gh-pages, @isaacabraham can you release docs? 
Or if you want I can try this time - I have mkdocs 1.0.4 so hopefully won't spoil the links anymore?

**Comment by mastoj** [comment]
@theimowski good, I was planning to share it on the internets but better to share when the docs have been published.

**Pull request merged**